src/engine/Server.cpp

```diff
@@ -113,14 +117,14 @@ boost::asio::awaitable<void> Server::process(
 
   if (params.contains("query")) {
     if (params.at("query").empty()) {
-      co_return co_await send(createBadRequestResponse(
+      co_return co_await sendWithCors(createBadRequestResponse(
           "Parameter \"query\" must not have an empty value", request));
     }
 
     co_return co_await processQuery(params, requestTimer, std::move(request),
                                     send);
```

Comment:
Also use `sendWithCors` here, then you don't need anything in the processQuery below.
(Currently, the TSV and CSV responses don't have their header fields changed,
those would also be included by this change).